n3fit/src/n3fit/model_trainer.py

```diff
@@ -523,12 +522,43 @@ def _generate_observables(
         self._reset_observables()
         log.info("Generating layers")
 
+        # We need to transpose Experimental data, stacking over replicas
+        experiment_data = {
+            "trmask": [],
+            "expdata": [],
+            "expdata_vl": [],
+            "invcovmat": [],
+            "invcovmat_vl": [],
+        }
+
+        # Loop over datasets
+        for i in range(len(self.exp_info[0])):
+            # Loop over data fields
+            for key in experiment_data.keys():
+                replica_data = []
+                # Loop over replicas
+                for replica in self.exp_info:
+                    if key in ["expdata", "expdata_vl"]:
+                        replica_data.append(replica[i][key].flatten())
```

Comment:
why do we need to flatten only expdata and exdata_vl ?

Reply:
Because for reasons beyond my understanding the expdata and expdata_vl are actually 2D arrays with a single row. Things would be simpler if the trvl mask were to be applied to the observation data too in the loss function, but that can be saved for a later iteration imo.